### DISCUSSION TITLE
COMSET succeeds callback

### DISCUSSION THREAD

**Discussion by sgouraud** [goToMain/libosdp#213] [Q&A]
As mentioned in your code, if COMSET succeeds, the PD must reply with the old params and then switch to the new params from then then on. As we first understood, the deprecated function osdp_set_command_complete_callback could be used for this. But after a library updated, this function was removed. Could you please explain the correct way to do this? Thanks in advance.

**Comment by sidcha** [comment]
You can do this by setting an event call back using [`osdp_cp_set_event_callback`](https://github.com/goToMain/libosdp/blob/master/include/osdp.h#L1130) and setting the the init-flag [`OSDP_FLAG_ENABLE_NOTIFICATION`](https://github.com/goToMain/libosdp/blob/master/include/osdp.h#L63) on the PD.

With that, you will recieve a new event type [`OSDP_EVENT_NOTIFICATION_COMMAND`](https://github.com/goToMain/libosdp/blob/master/include/osdp.h#L834) which reports the status of the last comset request.

**Reply by sgouraud** [reply]
Hi Siddharth,

Thanks for your reply :)

But does this also apply to PD devices? osdp_cp.c file is not included in our build system (we currently use the CONFIG_OSDP_STATIC_PD flag when compiling), sorry for not being clear about this.
 
We initially planed to add a command_callback after pd_send_reply if reply_id is REPLY_COM. Do you think there is a better way?

Thanks in advance,

Sam

**Reply by sidcha** [reply]
Ah, I see what you mean now. Indeed, there is no way for the PD app to receive a cue to switch to the new settings. Will see what can be done about that.

**Reply by sgouraud** [reply]
Here is a PR with my code modification : https://github.com/goToMain/libosdp/pull/215. Please feel free to comment, or maybe you see a better way to achieve that.

**Reply by sidcha** [reply]
Thanks for the PR. I am bit occupied ATM. Will get to this as soon as I get some free time.

**Comment by sidcha** [comment]
This change was merged as 8b8fac1451ede3fcbb13101c209b3d6e4a2ead33.
